src/standup/standup.service.ts

```diff
@@ -1,63 +1,69 @@
+import { AstraService, documentId } from '@cahllagerfeld/nestjs-astra';
 import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
 import { StandupDTO } from './dto/standup.dto';
 import { Standup } from './interfaces/standup.interface';
+import { tap } from 'rxjs/operators';
 
 @Injectable()
 export class StandupService {
-  private standups: Array<Standup> = [];
+  constructor(private readonly astraService: AstraService) {}
 
-  create(body: StandupDTO): Standup {
+  create(body: StandupDTO) {
     const { discordUser, todayMessage, yesterdayMessage } = body;
 
     const newStandup: Standup = {
-      id: 123,
       yesterdayMessage: yesterdayMessage,
       todayMessage: todayMessage,
       discordUser: discordUser,
-      createdOn: new Date('2021-01-01T00:00:00.000Z'),
+      createdOn: new Date(Date.now()),
     };
 
-    this.standups.push(newStandup);
-
-    return newStandup;
+    // this.standups.push(newStandup);
+    return this.astraService.create<Standup>(newStandup).pipe(
+      tap((data: documentId) => {
+        if (data === null) {
+          throw new HttpException(
+            'Creation didnt pass as expected',
+            HttpStatus.BAD_REQUEST,
+          );
+        }
+      }),
+    );
   }
 
-  findAll(): Array<Standup> {
-    return [...this.standups];
+  findAll() {
+    return this.astraService.find<Standup>();
   }
 
-  findById(id: number): Standup {
-    const result = this.standups.find((standup) => standup.id === id);
-    if (!result) {
-      throw new HttpException('Standup not found', HttpStatus.NOT_FOUND);
-    }
-
-    return { ...result };
+  findById(id: string) {
+    return this.astraService.get<Standup>(id).pipe(
+      tap((data: Standup) => {
+        if (data === null) {
+          throw new HttpException(
+            `no standup for ${id} found`,
+            HttpStatus.NOT_FOUND,
+          );
+        }
+      }),
+    );
   }
 
-  deleteStandup(id: number) {
-    const deleteElement = this.standups.find((standup) => standup.id == id);
-    if (!deleteElement) {
-      throw new HttpException('Standup not found', HttpStatus.NOT_FOUND);
-    }
-    const updateStandup = this.standups.filter((standup) => standup.id !== id);
-    this.standups = [...updateStandup];
-
-    return {};
+  deleteStandup(id: string) {
+    return this.astraService.delete(id);
```

Comment:
This is a huge problem: 
```ts
 async delete(path) {
    const response = await this.restClient.delete(`${this.basePath}/${path}`);
    if (response.status === 204) {
      return response.data;
    }
    return null;
  }
}
```
Thats the coding from the `@astrajs/collections` - package. If the response code isnt `204` null is returned.
Problem is: 
![image](https://user-images.githubusercontent.com/43843195/116915581-7bd54b00-ac4c-11eb-8b3d-28c64d59feae.png)
If the response is 204, stargate returns a `null`- Response as well. Thats why we cant react if the deletion is successfull or not 😅

Reply:
Ah that's a shame! Can we raise an issue on their repo?

Reply:
https://github.com/datastax/astrajs/issues/9

Reply:
Thanks 👍